src/components/TableX.vue

```diff
@@ -1,48 +1,104 @@
-<script lang="ts">
-/*Declearations had to be separated because of volar error*/
+<script setup lang="ts">
+import { onMounted, onUnmounted, ref } from 'vue';
 
-type TableCol = {
+export type TableCol = {
     label: string,
     name: string,
     className?: string,
     thClass?: string,
     width?: string
 };
-interface TableXProps{
+
+export interface TableXProps {
     cols: TableCol[],
     data: any[]
 }
-</script>
 
-<script setup lang="ts">
-const props = defineProps<TableXProps>();
-/*
-props.cols.forEach((col, idx)=>{
-    if(typeof col === 'string'){
-        props.cols[idx] = {
-            label: col
-        }
+let initialX = 0, initialWidth = 0, activeEl:HTMLTableCellElement;
+function drag(evt: MouseEvent){
+    const distance = evt.x - initialX;
+    activeEl.style.width = (initialWidth + distance) + 'px';
+}
+
+function dragStart(this:HTMLDivElement, evt: any){
+    initialX = evt.x;
+    activeEl = this.parentElement as HTMLTableCellElement;
+
+    initialWidth = activeEl.getBoundingClientRect().width;
+
+    window.addEventListener('mousemove', drag);
+}
+
+function dragEnd(){
+    if(initialX > 0){
+        initialX = 0;
+        window.removeEventListener('mousemove', drag);
     }
-})
-*/
+}
+
+const tableRef = ref<HTMLTableElement>();
+
+withDefaults<TableXProps, {}>(defineProps<TableXProps>(), {
+    width: 'auto'

```

Comment:
This line has an issue. I suppose `width: 'auto'` is meant as the default value for objects in `props.col` but this does not achieve that. In fact, I don't see this being used anywhere in the code